File: app/src/main/java/com/aircraft/toolmanagment/data/entity/BorrowReturnRecord.java

package com.aircraft.toolmanagment.data.entity;

import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "borrow_return_records")
public class BorrowReturnRecord {
    // Room 需要无参构造函数
    public BorrowReturnRecord() {}
    @PrimaryKey(autoGenerate = true)
    public int id;
    public int toolId;
    public int borrowerId;
    public long borrowTime;
    public long expectedReturnTime;
    public Long actualReturnTime;
    public String borrowReason;
    public String approvalStatus;
    public String rejectionReason;
}

File: app/src/main/java/com/aircraft/toolmanagment/data/entity/BorrowReturnRecord.kt
package com.aircraft.toolmanagment.data.entity

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "borrow_return_records")
data class BorrowReturnRecord(
    @PrimaryKey(autoGenerate = true)
    val id: Int = 0,
    val toolId: Int = 0,
    val borrowerId: Int = 0,
    val borrowTime: Long = 0,
    val expectedReturnTime: Long = 0,
    val actualReturnTime: Long? = null,
    val borrowReason: String = "",
    val approvalStatus: String = "",
    val rejectionReason: String = ""
)